### PR TITLE
Revert "SwiftDriver: initial work to properly handle android cross-compilation"

### PR DESCRIPTION
**Pull request by artemcm**
Reverts swiftlang/swift-driver#1560

Tentative revert to see if it resolves PR testing on the main repo.

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

File: Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift [hunk 1]
```diff
@@ -147,24 +147,14 @@ extension GenericUnixToolchain {
         }
       }
 
-      if targetInfo.sdkPath != nil {
-        for libpath in targetInfo.runtimeLibraryImportPaths {
-          commandLine.appendFlag(.L)
-          commandLine.appendPath(VirtualPath.lookup(libpath.path))
```
Comment (finagolfin) [inline Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift:153]: This is adding new runtime library paths to all platforms when `-sdk` is specified, which could be causing build errors.

**Comment by hborla** [comment]
@swift-ci please test Linux

**Review by hborla — APPROVED** [review]

**Comment by hborla** [comment]
Merging to unblock PR testing

**Pull request merged**